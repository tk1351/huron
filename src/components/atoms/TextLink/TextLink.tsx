import { FC } from 'react'

type Props = {
  title: string
  href: string
}

export const TextLink: FC<Props> = ({ title, href }) => (
  <a href={href}>{title}</a>
)
