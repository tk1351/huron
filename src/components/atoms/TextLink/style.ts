import { css } from '@emotion/react'
import { TextDecoration } from '../../../constants/types'

type TextLinkProps = {
  textDecoration: TextDecoration
}

export const link = ({ textDecoration }: TextLinkProps) =>
  css({
    textDecoration
  })
