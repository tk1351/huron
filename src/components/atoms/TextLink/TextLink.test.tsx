import * as renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react'
import { TextLink } from './TextLink'

const expectedTitle = 'textLink'
const expectedHref = '#'

const Component = <TextLink title={expectedTitle} href={expectedHref} />

describe('Testing TextLink Component', () => {
  describe('Render', () => {
    it('正しくTextLinkコンポーネントが表示されていること', () => {
      const App = renderer.create(Component)
      expect(App).toMatchSnapshot()
    })
  })

  describe('Props', () => {
    describe('title', () => {
      it('リンクのタイトルが正しく渡っていること', () => {
        render(Component)
        expect(screen.getByRole('link')).toHaveTextContent(expectedTitle)
      })
    })

    describe('href', () => {
      it('リンクのhrefに指定した値が正しく渡っていること', () => {
        render(Component)
        expect(screen.getByRole('link')).toHaveAttribute('href', expectedHref)
      })
    })
  })
})
