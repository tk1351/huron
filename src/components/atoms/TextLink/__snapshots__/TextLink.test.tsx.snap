// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextLink Component Render 正しくTextLinkコンポーネントが表示されていること 1`] = `
<a
  href="#"
>
  textLink
</a>
`;
